src/decoder.rs

```diff
@@ -51,6 +51,7 @@ impl <'a> Decoder<'a> {
     // If opts is None, make it empty Dict, otherwise take it as PyDict
     let opts1 = helpers::maybe_dict(py, opts);
     let aopt = helpers::get_atom_opt(py, &opts1)?;
+    let cached_atom_pyclass = opts1.get_item(py, "atom_call");
```

Comment:
@kvakvs I'm not 100% sure but I don't think I need a the `clone_ref` here right?

Reply:
Made some test, and this code doens't make any difference:
```
    let cached_atom_pyclass = match opts1.get_item(py, "atom_call") {
      Some(o) => Some(o.clone_ref(py)),
      None => None
    };
```